### DISCUSSION TITLE
Can Modbus func2 read 1 byte ?

### DISCUSSION THREAD

**Discussion by kagnadin** [artisan-roaster-scope/artisan#634] [Q&A]
I got a modbus rtu realy 2ch from aliexpress. 
http://www.chinalctech.com/m/view.php?aid=455
2ch output relay works very well using modbus rtu command, Drum & Cooler works well.

it has 2 optocouler input then modbus func2 is possible. but it supports only 1 byte, 08
if send 04 or 10(2bytes)  it doesn't relay.

11, Read optocoupler input status
send : FF 02 00 00 00 **08** 6C 12  <- 08 means 1 byte.  it works only at 08 ( coils request )
return : FF 02 01 01 51 A0
remarks:The 4th byte of the Return frame, Bit0--Bit7 of 0x01 represent  optocoupler1- optocoupler 8 input signal ,0 represent low level ,1 represent high level

In Artisan, modbus 1-2  for temperature( Autonics Tx4s, func4) ,  modbus 3-4 for optocouler.
Is there any way to send only 1byte( coils request : 08)  throught func2?

**Comment by fcebreros** [comment]
Hi,
not sure what you want to accomplish but to activate/deactivate the relays
it seems you have to use function 5.
So in artisan it would be

wcoil(slaveId,register,<bool>)

I think the default slave add is 255 and rly1 register is 0 for your board.

To read the status use

read(slaveID,register)


Best

Frank

**Reply by kagnadin** [reply]
Hi~  I want to use these as a switch to active some event button through the alarm. 
slave id doesn't matter. easay to chagne. 

At Config -> Ports Configuration  -> modbus tab 
There are 8 inputs  in inputs there are 1,2,3,4 function  but Artisan's Functions limits data length, 
So, I select func 2 then.. no options limits 1byte. 
![ex](https://user-images.githubusercontent.com/86217750/123056303-bcc63080-d441-11eb-9df7-26ec659c1784.png)

**Reply by fcebreros** [reply]
HI,
on this tab you only have to configure the modbus port. The input section is to read incoming data.
To configure the output relays you have to add buttons from the Config/events menu.
![image](https://user-images.githubusercontent.com/56903027/123125475-29273c80-d40e-11eb-9f9f-70fea3895401.png)
Here you can add buttons that send modbus commands.
